.github/CONTRIBUTING.md

```diff
@@ -91,24 +105,36 @@ The stylesheet available for use within DM can be found in `code/stylesheet.dm`.
 In order to make `Exited()` and `Entered()` procs more reliable, the usage of `forceMove()` when forcibly moving one item to another location, be it another item or turf, is required. Directly changing an item's loc values will skip over calls to the aforementioned procs, thus making them less useful and more unreliable.
 
 An example of improper item moving:
-```
+```DM
 /proc/some_proc(var/obj/A, var/obj/B)
 	A.loc = B	// Simply move A inside B.
 ```
 
 An example of proper item moving:
-```
+```DM
 /proc/some_proc(var/obj/A, var/obj/B)
 	A.forceMove(B)	// This will call A.loc.Exited() and B.Entered().
 					// The first method does not call either of those.
 ```
 
+### Regarding the variable usr
+If at all possible, procs outside of verbs and `Topic()` should avoid reliance on `usr`, and instead use a custom argument to specify the user and its expected type. This makes it easier to reuse procs in chains where `usr` is not always defined, and combats against potential security flaws that relying on `usr` can bring.
+
+`usr` must also always be validated to be the expected type!
+
+### Reserved argument names
+All BYOND procs have a set list of variables which are implicitly defined in each proc. However, the DM compiler will allow you to reuse these names as names for your custom variables. This should be avoided at all costs, to improve the readability and understanding of code.
+
+A list of reserved argument names instantiated with all procs:
+* `vars` being an alias of `src.vars` if `src` exists.
+* `args`
+* `usr`
+* `src`
+
+### Avoid `in world` loops
+Due to the amount of instances in the world, utilizing an `in world` loop should be avoided in all instances. In large projects, specially when crawling for types that are not optimized internally by BYOND (only core built-in types are), crawling `in world` will surmount to a very large blocking loop and will almost certainly trigger the infinite loop warning. This forces the specific callstack to sleep at least once.
```

Comment:
Maybe mention implicit in-worlds here too?

Reply:
addressed